source/server/filter_chain_factory_context_callback.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/api/v2/listener/listener.pb.h"
+#include "envoy/common/pure.h"
+#include "envoy/server/filter_config.h"
+
+namespace Envoy {
+namespace Server {
+
+/**
+ * Handles FilterChainFactoryContext creation. It is used by listener to adding new filter chain
+ * without worrying about the lifetime of each factory context.
+ */
+class FilterChainFactoryContextCreator {
+public:
+  virtual ~FilterChainFactoryContextCreator() = default;
+
+  /**
+   * Generate the filter chain factory context from proto. Note the callback will own the filter
+   * chain context.
```

Comment:
This comment about ownership is not correct right now, or at least it's not clear. I would rephrase as "Note the caller does not own the filter chain context."

Reply:
done